src/Database/Query.php

```diff
@@ -229,6 +236,40 @@ public function getConnection(): Connection
         return $this->_connection;
     }
 
+    /**
+     * Overrides the default connection role.
+     *
+     * @param string $role Connection role
+     * @return $this
+     */
+    public function setConnectionRole(string $role)
```

Comment:
Would be nice to have a `useReplica()` and `usePrimary()` methods on the query builder. That would save users having to import `Connection` to access the `ROLE_` constants, and remove the need to validate parameters.